drivers/i3c/i3c_mcux.c

```diff
@@ -427,37 +406,24 @@ static inline void mcux_i3c_status_clear_all(I3C_Type *base)
  *
  * @param base Pointer to controller registers.
  * @param mask Bits to be cleared.
- * @param init_delay_us Initial delay in microsecond before reading register
- *                      (can be 0).
- * @param step_delay_us Delay in microsecond between each read of register
- *                      (cannot be 0).
- * @param total_delay_us Total delay in microsecond before bailing out.
+ * @param timeout_us Timeout in microsecond before bailing out.
  *
  * @retval 0 If bits are cleared before time out.
- * @retval -ETIMEDOUT Exhausted all delays.
+ * @retval -ETIMEDOUT
  */
 static inline int mcux_i3c_status_clear_timeout(I3C_Type *base, uint32_t mask,
-						uint32_t init_delay_us,
-						uint32_t step_delay_us,
-						uint32_t total_delay_us)
+						uint32_t timeout_us)
 {
-	uint32_t delayed = init_delay_us;
-	int ret = -ETIMEDOUT;
-
-	/* Try to clear bit until it is cleared */
-	while (delayed <= total_delay_us) {
-		base->MSTATUS = mask;
-
-		if (!mcux_i3c_status_is_set(base, mask)) {
-			ret = 0;
-			break;
-		}
-
-		k_busy_wait(step_delay_us);
-		delayed += step_delay_us;
+	base->MSTATUS = mask;
+	/*
+	 * Status should clear quickly so no extra delays between
+	 * checks. Use the delay_stmt to retry clearing the
+	 * status by writing to the MSTATUS register.
+	 */
+	if (!WAIT_FOR(!mcux_i3c_status_is_set(base, mask), timeout_us, base->MSTATUS = mask)) {
```

Comment:
`==`?

Reply:
As I wrote in the comment, it's intentionally using the delay_stmt field of the WAIT_FOR() macro to write to the STATUS register, like the original code was doing. I'll use a temp variable to get it out of the if() statement to avoid the checkpatch false positive.